### PR TITLE
SingleCrystalDiffuseReduction updates

### PR DESCRIPTION
**Pull request by rosswhitfield**
* SingleCrystalDiffuseReduction has been update to use MDNorm instead of MDNormSCD internally.
* Additionally more options have been added to apply either a calibration with ApplyCalibration or to copy an Instrument with CopyInstrumentParameters (these were also added to ConvertMultipleRunsToSingleCrystalMD)
* Options have been added that allow you to specify either a UB matrix file or omega offset separately for each run
* By default the SolidAngle and Flux workspaces will not be deleted and will be reused the next time the algorithm is used.
* Incompatible changes include changing of parameters names for projection, binning and symmetry operations to match MDNorm; symmetry operations will now use the symmetry of the point group instead of space group and will no longer accept space group number to avoid ambiguity of which point group to use; binning parameter has been changed match MDNorm where the bin width is specified instead of the number of bins.

**To test:**

Try the SingleCrystalDiffuseReduction usage examples.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SingleCrystalDiffuseReduction.py [hunk 1]
```diff
@@ -197,134 +215,112 @@ def PyExec(self):
...
-                              Axis0=self.getProperty('Axis0').value,
-                              Axis1=self.getProperty('Axis1').value,
-                              Axis2=self.getProperty('Axis2').value)
+        for n, run in enumerate(self.getProperty("Filename").value):
+            logger.notice("Working on " + run)
+
+            self.load_file_and_apply(run, '__run', _offsets[n])
+            LoadIsawUB('__run', _UB[n])
+
+            MinValues, MaxValues = ConvertToMDMinMaxGlobal(InputWorkspace='__run',
```
Comment (AndreiSavici) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SingleCrystalDiffuseReduction.py]: I just realized that `MinValues` are all `-self.XMax` and `MaxValues` are all `self.XMax`

Comment (rosswhitfield) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SingleCrystalDiffuseReduction.py]: So it is. I will change this.

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SingleCrystalDiffuseReduction.py [hunk 2]
```diff
@@ -197,134 +214,110 @@ def PyExec(self):
...
-            MinusMD(LHSWorkspace=_outWS_name+'_normalizedData',RHSWorkspace='__scaled_background',OutputWorkspace=_outWS_name)
+            MinusMD(LHSWorkspace='__normalizedData',RHSWorkspace='__normalizedBackground',OutputWorkspace=_outWS_name)
             if self.getProperty('KeepTemporaryWorkspaces').value:
                 RenameWorkspaces(InputWorkspaces=['__data','__norm','__bkg_data','__bkg_norm'],
                                  WorkspaceNames=[_outWS_name+'_data', _outWS_name+'_normalization',
                                                  _outWS_name+'_background_data',_outWS_name+'_background_normalization'])
         else:
             # outWS = data / norm
-            DivideMD(LHSWorkspace='__data',RHSWorkspace='__norm',OutputWorkspace=_outWS_name)
+            RenameWorkspace(InputWorkspace='__normalizedData',OutputWorkspace=_outWS_name)
```
Comment (AndreiSavici) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SingleCrystalDiffuseReduction.py]: You could set directly the output workspace name in `MDNorm` for data, then optionally for `MinusMD` if you have background, so no need to rename

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SingleCrystalDiffuseReduction.py [hunk 3]
```diff
@@ -57,10 +63,12 @@ def PyInit(self):
         self.copyProperties('LoadEventNexus', ['FilterByTofMin', 'FilterByTofMax'])
 
         # Vanadium SA and flux
+        self.declareProperty("ReuseSAFlux", True, "If True then if a previous SolidAngle and Flux has been loaded "
+                             "it will be reused otherwise it will be loaded.")
         self.declareProperty(FileProperty(name="SolidAngle",defaultValue="",action=FileAction.Load,
                                           extensions=[".nxs"]),
                              doc="An input workspace containing momentum integrated vanadium (a measure"
-                             "of the solid angle). See :ref:`MDnormSCD <algm-MDnormSCD>` for details")
+                             "of the solid angle). See :ref:`MDNormSCD <algm-MDNormSCD>` for details")
```
Comment (AndreiSavici) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SingleCrystalDiffuseReduction.py]: I would replace `MDNormSCD` with `MDNormSCDPreprocessIncoherent` (optional)

File: docs/source/algorithms/SingleCrystalDiffuseReduction-v1.rst [hunk 4]
```diff
@@ -61,15 +62,14 @@ If no background is used then the "ws" is just the normalised data.
...
+:ref:`point or space group <Point and space groups>` were all the
+symmetries for that group is applied, or you can specify individual
+:ref:`symmetries <Symmetry groups>` to apply.
 
-For example setting SymmetryOps to "P 31 2 1", "152" or "x,y,z;
--y,x-y,z+1/3; -x+y,-x,z+2/3; y,x,-z; x-y,-y,-z+2/3; -x,-x+y,-z+1/3"
-are equivalent.
+For example setting SymmetryOperations to space group "P 31 2 1",
+point group "321" or "x,y,z; -y,x-y,z; -x+y,-x,z; y,x,-z; x-y,-y,-z;
+-x,-x+y,-z" are equivalent.
```
Comment (AndreiSavici) [inline docs/source/algorithms/SingleCrystalDiffuseReduction-v1.rst]: Maybe I should have added to mine as well, but to clarify any confusion, these are symmetries of the real lattice, not of the reciprocal lattice.

File: docs/source/algorithms/SingleCrystalDiffuseReduction-v1.rst [hunk 5]
```diff
@@ -227,13 +226,18 @@ Usage
...
-                                 SymmetryOps="P 31 2 1")
+                                 QDimension0='1,1,0',
+                                 QDimension1='1,-1,0',
+                                 QDimension2='0,0,1',
+                                 Dimension0Binning='-7.5375,0.075,7.5375',
+                                 Dimension1Binning='-13.165625,0.13100125,13.165625',
+                                 Dimension2Binning='-0.1,0.1',
+                                 SymmetryOperations="P 31 2 1")
+
+   fig, ax = plt.subplots(subplot_kw={'projection':'mantid'})
```
Comment (AndreiSavici) [inline docs/source/algorithms/SingleCrystalDiffuseReduction-v1.rst]: add 
```python
import matplotlib.pyplot as plt
from mantid import plots
```

**Review by AndreiSavici — CHANGES_REQUESTED** [review]
The workflow diagram is out of date. Also, the related algorithm at the end should point to `MDNorm`

**Review by AndreiSavici — APPROVED** [review]

**Pull request merged**